ha/core/system_health/system_health.py

```diff
@@ -402,41 +427,72 @@ def process_event(self, healthevent: HealthEvent):
             component_id = healthevent.resource_id
             Log.info(f"SystemHealth: Processing {component}:{component_type}:{component_id} with status {status}")
 
+            # Update the node map
+            self.node_id = healthevent.node_id
+            self.node_map = {'cluster_id':healthevent.cluster_id, 'site_id':healthevent.site_id,
+                             'rack_id':healthevent.rack_id, 'storageset_id':healthevent.storageset_id}
+
             # Read the currently stored health value
             current_health = self.get_status_raw(component, component_id, comp_type=component_type,
                                         cluster_id=healthevent.cluster_id, site_id=healthevent.site_id,
                                         rack_id=healthevent.rack_id, storageset_id=healthevent.storageset_id,
                                         node_id=healthevent.node_id, server_id=healthevent.node_id,
                                         storage_id=healthevent.node_id)
 
+            current_timestamp = str(int(time.time()))
             if current_health:
+                current_health_dict = json.loads(current_health)
+                specific_info = current_health_dict["events"][0]["specific_info"]
+            if current_health and specific_info:
+                # If health is already stored and its a node_health, check further
+                stored_genration_id = current_health_dict["events"][0]["specific_info"]["generation_id"]
+                incoming_generation_id = healthevent.specific_info["generation_id"]
+                incoming_health_status = current_health_dict["events"][0]["status"]
+                pod_restart_val = current_health_dict["events"][0]["specific_info"]["pod_restart"]
                 # Update the current health value itself.
-                updated_health = EntityHealth.read(current_health)
+                latest_health = EntityHealth.read(current_health)
+                if stored_genration_id != incoming_generation_id:
```

Comment:
This additional logic is only applicable to "node" events. 
I think we should add a check for resource_type="node" where additional events are being sent /handled ? 
Otherwise for hierarchical updates also this logic will execute, causing potential issues ?

Reply:
I think 
 resource_type="node" checking is not required. This will be handled in else part. (Which I talked about in above comment). I am updating the PR with this